test/app/core/managers/audio_record_services_test.dart

```diff
@@ -0,0 +1,313 @@
+        when(() => soundRecorder.isStopped).thenReturn(false);
+        when(() => audioSyncManager.syncAudio())..thenAnswer((_) async => true);
+        when(() => soundRecorder.stopRecorder()).thenAnswer((_) async => null);
+        when(() => soundRecorder.closeRecorder()).thenAnswer((_) async => null);
+
+        // act
+        await recordServices.dispose();
+
+        // assert
+        verify(() => soundRecorder.isStopped).called(1);
+        verify(() => soundRecorder.stopRecorder()).called(1);
+        verify(() => soundRecorder.closeRecorder()).called(1);
+        verify(() => audioSyncManager.syncAudio()).called(1);
+      },
+    );
+
+    test(
+      'start() record audio for microphone permission granted',
+      () async {
+        // arrange
+        final mockPermissionHandlerPlatform =
+            PermissionHandlerPlatform.instance;
+        debugDefaultTargetPlatformOverride = TargetPlatform.iOS;
+
+        when((() => mockPermissionHandlerPlatform
+                .checkPermissionStatus(Permission.microphone)))
+            .thenAnswer((_) async => PermissionStatus.granted);
+
+        when(() => soundRecorder.isStopped).thenReturn(true);
+        when(() => soundRecorder.openRecorder()).thenAnswer((_) async => null);
+        when(() => soundRecorder.setSubscriptionDuration(any()))
+            .thenAnswer((_) async => null);
+        when(() => soundRecorder.startRecorder(
+              codec: any(named: 'codec'),
+              toFile: any(named: 'toFile'),
+              bitRate: any(named: 'bitRate'),
+              sampleRate: any(named: 'sampleRate'),
+            )).thenAnswer((_) async => null);
+
+        when(() => audioSyncManager.audioFile(
+              session: any(named: 'session'),
+              sequence: any(named: 'sequence'),
+            )).thenAnswer((_) async => '/tmp/file_name.txt');
+
+        // act
+        await recordServices.start();
+
+        // assert
+        verify(() => soundRecorder.isStopped).called(1);
+        verify(() => soundRecorder.closeRecorder()).called(1);
+        verify(() => soundRecorder.openRecorder()).called(1);
+        verify(() => soundRecorder.startRecorder(
+            codec: any(named: 'codec'),
+            toFile: any(named: 'toFile'),
+            bitRate: any(named: 'bitRate'),
+            sampleRate: any(named: 'sampleRate'))).called(1);
+      },
+    );
+
+    test(
+      'start() do not crash if soundRecorder crash',
+      () async {
+        // arrange
+        final mockPermissionHandlerPlatform =
+            PermissionHandlerPlatform.instance;
+        debugDefaultTargetPlatformOverride = TargetPlatform.iOS;
+
+        when((() => mockPermissionHandlerPlatform
+                .checkPermissionStatus(Permission.microphone)))
+            .thenAnswer((_) async => PermissionStatus.granted);
+
+        when(() => soundRecorder.isStopped).thenReturn(true);
+        when(() => soundRecorder.openRecorder()).thenAnswer((_) async => null);
```

Comment:
O retorno do método `flutterSoundPlayer.openPlayer()` é `Future<void>`, nunca retornará nulo, acha que faria sentido o mock ser `void` também?
```suggestion
        when(() => soundRecorder.openRecorder()).thenAnswer((_) async {});
```